scripts/update-metadata.mjs

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env node
+
+import fs from "node:fs";
+import path from "node:path";
+import { fileURLToPath } from 'node:url';
+
+const CROWDIN_PROJECT_ID = process.env.CROWDIN_PROJECT_ID;
+const CROWDIN_PERSONAL_TOKEN = process.env.CROWDIN_PERSONAL_TOKEN;
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const rootDir = path.resolve(__dirname, '..');
+const metadataFile = path.join(rootDir, 'crowdin-metadata.json');
+
+const headers = {
+    'Authorization': `Bearer ${CROWDIN_PERSONAL_TOKEN}`,
+    'Content-Type': 'application/json'
+};
+
+const url = `https://api.crowdin.com/api/v2/projects/${CROWDIN_PROJECT_ID}/languages/progress?limit=500`;
+
+// All changed files passed in as CLI arguments (after index 2 because index 0 is node and index 1 is the script name)
+const changedFiles = process.argv.slice(2);
+
+async function run() {
+    // Build a payload containing updates for locales that had changes
+    let diffData = await getDiffPayload(changedFiles);
+    // Read the current metadata from the file
+    const existingData = getExistingMetadata();
+
+    // Merge existing metadata with the new diff data
+    const mergedData = mergeData(existingData, diffData);
+
+    // Save updated metadata back to the file
+    const jsonData = JSON.stringify(mergedData, null, 2);
+    fs.writeFileSync(metadataFile, jsonData + '\n', 'utf8');
+}
+
+// Build an update payload for locales affected by changed files including their locale, last updatedAt and the percentage of translation progress
+async function getDiffPayload(changedFiles) {
+    const payload = [];
+    if (changedFiles.length === 0) return payload;
+
+    const processedLocales = new Set();
+    const crowdinData = await fetchCrowdinData();
+
+    for (const file of changedFiles) {
+        // Only process files inside the translations/ directory
+        if (!file.startsWith('translations/')) continue;
+
+        const pathParts = file.split('/');
+        if (pathParts.length < 2) continue;
+
+        const locale = pathParts[1];
+
+        // Skip if locale already handled in this run
+        if (processedLocales.has(locale)) continue;
+        processedLocales.add(locale);
+
+        // If we have Crowdin data for this locale, add it to the payload
+        if (crowdinData[locale]) {
+            payload.push({
+                locale: locale,
+                updatedAt: new Date().toISOString(),
```

Comment:
```suggestion
                updatedAt: new Date().getTime(), 
```

How about timestamps instead? What's more convenient to work with in php later on?